### PR TITLE
fix: rounding up

### PR DESCRIPTION
**Pull request by Roznovjak**
Do not round up return value in calculate_out_given_in and do not round up zero return value in calculate_in_given_out.

### CODE REVIEW

File: src/xyk.rs [hunk 1]
```diff
@@ -56,8 +56,7 @@ pub fn calculate_out_given_in(
     let numerator = out_reserve_hp.checked_mul(amount_in_hp).ok_or(Overflow)?;
     let sale_price_hp = numerator.checked_div(denominator).ok_or(Overflow)?;
 
-    let result = to_balance!(sale_price_hp).ok();
-    round_up!(result.ok_or(Overflow)?)
+    to_balance!(sale_price_hp)
```
Comment (enthusiastmartin) [inline src/xyk.rs:63]: I think we should add the same check of zero amount in this function too. 

Also add a test case for this.

**Review by enthusiastmartin — COMMENTED** [review]
looks ok.

File: src/xyk.rs (hunk 1 above)
Comment (Roznovjak) [reply src/xyk.rs:63]: done

**Review by enthusiastmartin — APPROVED** [review]
bump version and merge.

**Pull request merged**